src/Core/Framework/Gateway/Context/Command/Handler/RegisterCustomerCommandHandler.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Gateway\Context\Command\Handler;
+
+use Shopware\Core\Checkout\Customer\SalesChannel\AbstractRegisterRoute;
+use Shopware\Core\Framework\Gateway\Context\Command\AbstractContextGatewayCommand;
+use Shopware\Core\Framework\Gateway\Context\Command\RegisterCustomerCommand;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Validation\DataBag\DataBag;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\PlatformRequest;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+/**
+ * @extends AbstractContextGatewayCommandHandler<RegisterCustomerCommand>
+ */
+#[Package('framework')]
+class RegisterCustomerCommandHandler extends AbstractContextGatewayCommandHandler
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly AbstractRegisterRoute $registerRoute,
+    ) {
+    }
+
+    public function handle(AbstractContextGatewayCommand $command, SalesChannelContext $context, array &$parameters): void
+    {
+        /** @var array<string, mixed> $data */
+        $data = $command->data;
+        $data['billing'] = new DataBag($data['billingAddress']);
+
+        if (\array_key_exists('shippingAddress', $data) && \is_array($data['shippingAddress'])) {
+            $data['shipping'] = new DataBag($data['shippingAddress']);
+        }
```

Comment:
I don't understand this remapping.
- they are both optional, not just the shippingAddress
- RegisterRoute wants array-keys `billingAddress` and `shippingAddress`